src/Operations.jl

```diff
@@ -938,6 +938,8 @@ end
 function develop(ctx::Context, pkgs::Vector{PackageSpec}, new_git::Vector{UUID})
     # no need to look at manifest.. dev will just nuke whatever is there before
     for pkg in pkgs
+        get(ctx.env.project.deps, pkg.name, pkg.uuid) == pkg.uuid ||
+            pkgerror("`$(pkg.name)` already exists as a direct dependency with UUID `$(pkg.uuid)`.")
```

Comment:
Maybe be more explicit, e.g.
```suggestion
            pkgerror("cannot add package `$(pkg.name) = $(repr(pkg.uuid))` since package `$(other.name) =  $(repr(other.uuid))` already exists as a direct dependency.")
```

Reply:
Much better!